File: wheystore/wheystore.Web/Modules/SanPhamDB/QuaTang/QuaTangRow.cs
namespace wheystore.SanPhamDB;

[ConnectionKey("Default"), Module("SanPhamDB"), TableName("QuaTang")]
[DisplayName("Qua Tang"), InstanceName("Qua Tang")]
[ReadPermission("Administration:General")]
[ModifyPermission("Administration:General")]
[ServiceLookupPermission("Administration:General")]
public sealed class QuaTangRow : Row<QuaTangRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Id"), Identity, IdProperty]
    public int? Id { get => fields.Id[this]; set => fields.Id[this] = value; }

    [DisplayName("Ten Qua"), Size(200), NotNull, QuickSearch, NameProperty]
    public string TenQua { get => fields.TenQua[this]; set => fields.TenQua[this] = value; }

    [DisplayName("Gia Tri"), Size(18)]
    public decimal? GiaTri { get => fields.GiaTri[this]; set => fields.GiaTri[this] = value; }

    [DisplayName("Mo Ta")]
    public string MoTa { get => fields.MoTa[this]; set => fields.MoTa[this] = value; }

    [DisplayName("Anh"), Size(255)]
    public string Anh { get => fields.Anh[this]; set => fields.Anh[this] = value; }

    public class RowFields : RowFieldsBase
    {
        public Int32Field Id;
        public StringField TenQua;
        public DecimalField GiaTri;
        public StringField MoTa;
        public StringField Anh;

    }
}

File: wheystore/wheystore.Web/Modules/ServerTypes/SanPhamDB/QuaTangRow.ts
import { fieldsProxy } from "@serenity-is/corelib";

export interface QuaTangRow {
    Id?: number;
    TenQua?: string;
    GiaTri?: number;
    MoTa?: string;
    Anh?: string;
}

export abstract class QuaTangRow {
    static readonly idProperty = 'Id';
    static readonly nameProperty = 'TenQua';
    static readonly localTextPrefix = 'SanPhamDB.QuaTang';
    static readonly deletePermission = 'Administration:General';
    static readonly insertPermission = 'Administration:General';
    static readonly readPermission = 'Administration:General';
    static readonly updatePermission = 'Administration:General';

    static readonly Fields = fieldsProxy<QuaTangRow>();
}